src/components/header/header.js

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+
+
+export default class Header extends React.Component {
+    constructor () {
+        super()
+
+        this.state = {
+            text : ' '
+        };
+
+        this.pressEnter = this.pressEnter.bind(this);
```

Comment:
binding is actually not needed if you use arrow functions, so you can just remove this bunding for now and see my comments below